app/services/email_normalizer.rb

```diff
@@ -0,0 +1,41 @@
+require 'resolv'
+
+# Class to help normalize email addresses for services like Gmail that let users
+# add extra things after a +
+class EmailNormalizer
+  attr_reader :email
+
+  # @param [#to_s] email
+  def initialize(email)
+    @email = Mail::Address.new(email.to_s)
+  end
+
+  # @return [String]
+  def normalized_email
+    if gmail?
+      before_plus, _after_plus = email.local.split('+', 2)
+      [before_plus.tr('.', ''), email.domain].join('@')
+    else
+      email.to_s
+    end
+  end
+
+  private
+
+  def gmail?
+    email.domain == 'gmail.com' || google_mx_record?
+  end
+
+  def google_mx_record?
+    return false if ENV['RAILS_OFFLINE']
```

Comment:
mimicking the behavior we have for the `email_validator` gem so that CI and other test environments don't have to talk to the real internet

https://github.com/18F/identity-idp/blob/de5fa81435aed726fdca831fc4889e6611e2b2c2/app/validators/form_email_validator.rb#L10-L14